homeassistant/components/idasen_desk/entity.py

```diff
@@ -0,0 +1,32 @@
+"""Base entity for Idasen Desk."""
+
+from __future__ import annotations
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IdasenDeskCoordinator
+
+
+class IdasenDeskEntity(CoordinatorEntity[IdasenDeskCoordinator]):
+    """IdasenDesk sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        unique_id: str,
+        device_info: DeviceInfo,
+        coordinator: IdasenDeskCoordinator,
+    ) -> None:
+        """Initialize the IdasenDesk sensor entity."""
+        super().__init__(coordinator)
+
+        self._attr_unique_id = unique_id
+        self._attr_device_info = device_info
```

Comment:
Might as well create the device here instead of earlier

Reply:
`IdasenDeskButton` does not use this class, so it would be duplicating the DeviceInfo init

Reply:
Why doesn't it use this class?

Reply:
The buttons should not change to unavailable when the device disconnects, since the buttons are used to connect/disconnect. 
Should I still make them inherit from this and just override `available()` there?

Reply:
done. let me know if its better now

Reply:
I would overwrite available to always return True.